### PR TITLE
idx: support security question authenticator - OKTA-448291

### PR DESCRIPTION
**Pull request by shuowu**

### CODE REVIEW

**Review by denysoblohin-okta — APPROVED** [review]

File: lib/idx/authenticator/GeneralAuthenticator.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Authenticator } from './Authenticator';
+
+export type GeneralAuthenticatorInputValues = {
+  verificationCode: string;
+};
+
+export class GeneralAuthenticator extends Authenticator {
```
Comment (jaredperreault-okta) [inline lib/idx/authenticator/GeneralAuthenticator.ts:7]: maybe rename this `PasscodeAuthenticator`? `GeneralAuthenticator` sort sounds like another base class

File: samples/generated/static-spa/public/index.html [hunk 2]
```diff
@@ -2,14 +2,14 @@
 <html>
   <head>
     <!-- In this sample, okta-auth-js assets are being served locally from node_modules. Assets are also available on CDN -->
-    <!-- https://global.oktacdn.com/okta-auth-js/5.8.0/okta-auth-js.polyfill.js -->
-    <!-- https://global.oktacdn.com/okta-auth-js/5.8.0/okta-auth-js.min.js -->
+    <!-- https://global.oktacdn.com/okta-auth-js/5.9.1/okta-auth-js.polyfill.js -->
+    <!-- https://global.oktacdn.com/okta-auth-js/5.9.1/okta-auth-js.min.js -->
```
Comment (jaredperreault-okta) [inline samples/generated/static-spa/public/index.html:6]: delete these comments?

File: lib/idx/authenticator/GeneralAuthenticator.ts (hunk 1 above)
Comment (shuowu) [reply lib/idx/authenticator/GeneralAuthenticator.ts:7]: How about `VerificationCodeAuthenticator`? to match with the input name. I think I'll also need to add some comment there.

**Review by jaredperreault-okta — APPROVED** [review]

File: samples/generated/express-embedded-auth-with-sdk/web-server/views/authenticator copy.mustache [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
Comment (denysoblohin-okta) [inline samples/generated/express-embedded-auth-with-sdk/web-server/views/authenticator copy.mustache:1]: Do we need this file?

File: samples/templates/express-embedded-auth-with-sdk/web-server/routes/authenticator.js [hunk 4]
```diff
@@ -273,4 +273,48 @@ router.post('/challenge-authenticator/google_otp', async (req, res, next) => {
...
 
+// Handle Security Question authenticator
+router.get('/enroll-authenticator/security_question', async (req, res) => {
+  renderPage({
+    req, res,
+    render: () => renderTemplate(req, res, 'authenticator-security_question', {
+      req, res,
+      title: 'Enroll security question authenticator',
+      action: '/enroll-authenticator/security_question',
+      input: {
```
Comment (denysoblohin-okta) [inline samples/templates/express-embedded-auth-with-sdk/web-server/routes/authenticator.js:284]: Looks like `input ` is not correct here (should be array of `questionCode`, `question`, `answer`), right?
And it's not used in the template?

Comment (shuowu) [reply samples/templates/express-embedded-auth-with-sdk/web-server/routes/authenticator.js:284]: questionCode and question are handled in the mustache template.

File: samples/generated/express-embedded-auth-with-sdk/web-server/views/authenticator copy.mustache (hunk 3 above)
Comment (shuowu) [reply samples/generated/express-embedded-auth-with-sdk/web-server/views/authenticator copy.mustache:1]: Good catch! will regenerate the samples.

**Review by jaredperreault-okta — APPROVED** [review]

**Pull request closed without merging**